### PR TITLE
docs: #1653 scalable retention and manifest compression

### LINKED ISSUE #1653: Document scalable retention and manifest compression
A bit of (advanced) tunable conf is all we want to say in docs about these internal-facing scale improvements for Tiered Storage  in 23.2

Work tracked in:

https://github.com/redpanda-data/core-internal/issues/373
https://github.com/redpanda-data/core-internal/issues/389

SMEs: @jcsp  and @Lazin

### PR DESCRIPTION
**Pull request by micheleRP**
resolves #1653 

This adds a small conceptual section about archived metadata to the Tiered Storage pages (linux & k8s) as well as the How Redpanda Works page. 

preview:
- https://deploy-preview-1685--redpanda-documentation.netlify.app/docs/beta/manage/tiered-storage/#archived-metadata
- https://deploy-preview-1685--redpanda-documentation.netlify.app/docs/beta/get-started/architecture/

### CODE REVIEW

File: versioned_docs/version-23.2/get-started/architecture.mdx [hunk 1]
```diff
@@ -24,6 +24,8 @@ Consumers that read from more recent offsets continue to read from local storage
 
 To scale topics, Redpanda shards them into one or more partitions that are distributed across the nodes in a cluster. This allows for concurrent writing and reading from multiple nodes. When producers write to a topic, they route events to one of the topic’s partitions. Events with the same key (like a stock ticker) are always routed to the same partition, and Redpanda guarantees the order of events at the partition level. Consumers read events from a partition in the order that they were written. If a key is not specified, then events are sent to all topic-partitions in a round-robin fashion.
 
+To support efficient long-term data retention with Tiered Storage, Redpanda uses compressed and scalable partition STM manifests. When the partition STM manifest for object storage grows too large, Redpanda generates external manifests for archive storage. No memory or disk is consumed for those segments tracked in archive storage.
```
Comment (Lazin) [inline versioned_docs/version-23.2/get-started/architecture.mdx:27]: Do we ever mention manifests in the docs? As far as I'm concerned this is our internal implementation detail. The big change is that we have an "archive" now. So previously we had local data which is stored on disk on Redpanda node and also we had data in the cloud. Now the "cloud" part of the log is split into two parts. The main one for which we're storing metadata in Redpanda all the time and the "archive" for which we're not storing any metadata at all. "Archive" metadata is stored in the cloud.

My weak attempt to phrase this correctly for docs:
```
To support efficient long-term data retention with Tiered-Storage, we changed the way metadata is stored. Previously, we stored all metadata needed to access data in the cloud in Redpanda at any given moment in time. But now we're only storing metadata for recently uploaded segments. The rest of the metadata is safely archived in the cloud storage and cached locally on disk. This metadata can be loaded when the historical data is accessed. This allows Tiered-Storage in Redpanda to handle partitions of practically any size.
```

File: versioned_docs/version-23.2/get-started/architecture.mdx [hunk 2]
```diff
@@ -20,6 +20,8 @@ Redpanda [Tiered Storage](../../manage/tiered-storage) is a multi-tiered object
 
 Consumers that read from more recent offsets continue to read from local storage, and consumers that read from historical offsets read from object storage, all with the same API. Consumers can read and reread events from any point within the maximum retention period, whether the events reside on local or object storage.
 
+As data in object storage grows, the metadata for it grows. To support efficient long-term data retention, Redpanda splits the metadata in object storage, storing metadata of only recently-updated segments, and safely archiving the remaining metadata and caching it locally on disk. Archived metadata can be loaded when historical data is accessed. This allows Tiered Storage to handle partitions of virtually any size.
```
Comment (shorgi) [inline versioned_docs/version-23.2/get-started/architecture.mdx:23]: For `... safely archiving the remaining metadata and caching it locally on disk`, where is the metadata archived, in object storage? The sentence can be read as metadata being archived locally on disk.

Comment (micheleRP) [reply versioned_docs/version-23.2/get-started/architecture.mdx:23]: rewrote as `safely archiving the remaining metadata in object storage and caching it locally on disk. `

File: versioned_docs/version-23.2/manage/shared/tiered-storage/_external-manifests.mdx [hunk 3]
```diff
@@ -0,0 +1,3 @@
+As data in object storage grows, the metadata for it grows. To support efficient long-term data retention, Redpanda splits the metadata in object storage, storing metadata of only recently-updated segments, and safely archiving the remaining metadata and caching it locally on disk. Archived metadata can be loaded when historical data is accessed. This allows Tiered Storage to handle partitions of virtually any size.
+
+The size of the archive storage is configurable. The `cloud_storage_target_manifest_size_bytes` property contains the number of elements in archive storage. To access data in the archive, Redpanda uses a `ListObjectsV2` API request to fetch a list of external manifests, which are used to access individual segments.
```
Comment (mattschumpert) [inline versioned_docs/version-23.2/manage/shared/tiered-storage/_external-manifests.mdx:3]: - This page sits right above a page about 'data archiving' where the 'archive' is the data itself.  So here we should clarify this is about metadata by saying 'metadata archive storage is configurable'
- this property doesn't contain the 'number of elements' but rather sets the target size in bytes of the metadata archive in object storage.

**Review by mattschumpert — APPROVED** [review]

**Comment by micheleRP** [comment]
@Lazin Would you please give this a final review and approve it for engineering if it's good to publish in our Beta docs?

**Review by jcsp — APPROVED** [review]

**Pull request merged**